presto-spi/src/main/java/com/facebook/presto/spi/type/Chars.java

```diff
@@ -100,19 +101,32 @@ public static Slice trimSpacesAndTruncateToLength(Slice slice, int maxLength)
     public static Slice trimTrailingSpaces(Slice slice)
     {
         requireNonNull(slice, "slice is null");
-        return slice.slice(0, sliceLengthWithoutTrailingSpaces(slice));
+        return slice.slice(0, sliceLengthWithoutTrailingSpaces(slice, 0, slice.length()));
     }
 
-    private static int sliceLengthWithoutTrailingSpaces(Slice slice)
+    public static int sliceLengthWithoutTrailingSpaces(Slice slice, int offset, int length)
```

Comment:
`lengthAfterTrimmingTrailingSpaces`?

Reply:
`byteCountWithoutTrailingSpaces` (same for the one below)